x-pack/solutions/security/plugins/security_solution/public/cases/components/case_events/search_events.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EcsFlat } from '@elastic/ecs';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { AbortError } from '@kbn/kibana-utils-plugin/common';
+import type { SortColumnTable } from '@kbn/securitysolution-data-table';
+import type { EcsSecurityExtension } from '@kbn/securitysolution-ecs';
+import type { TimelineItem } from '@kbn/timelines-plugin/common';
+import { lastValueFrom } from 'rxjs';
+import { KibanaServices } from '../../../common/lib/kibana';
+
+export const searchEvents = async (
+  signal: AbortSignal | undefined,
+  dataView: DataView | undefined,
+  parameters: {
+    eventIds: string[];
+    sort: SortColumnTable[];
+    pageIndex: number;
+    itemsPerPage: number;
+  }
+): Promise<TimelineItem[]> => {
+  if (!dataView) {
+    throw new Error('data view is not defined');
+  }
+
+  const { data } = KibanaServices.get();
+
+  const response = await lastValueFrom(
+    data.search.search({
+      params: {
+        index: dataView.getIndexPattern(),
+        body: {
+          query: {
+            ids: {
+              values: parameters.eventIds,
+            },
+          },
+        },
+        fields: ['*'],
+        sort: parameters.sort.map((tableSort) => ({
+          [tableSort.columnId]: tableSort.sortDirection,
+        })),
+        from: parameters.pageIndex * parameters.itemsPerPage,
+        size: parameters.itemsPerPage,
+      },
+    })
+  );
+
+  if (signal?.aborted) {
+    throw new AbortError();
+  }
+
+  return (
+    response?.rawResponse?.hits?.hits?.map((row) => {
+      const ecs = structuredClone(EcsFlat) as unknown as EcsSecurityExtension;
```

Comment:
I think you can just do this without casting

```
 const ecs = {
        _id: row?._id as string,
        _index: row._index as string,
};
```

Reply:
thanks, seems like copying the ecs is not needed for table to render